optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -240,4 +255,170 @@ public void skipAhead() throws ExecutionException, InterruptedException {
+            String solutionName = Integer.toString(id);
+            if (onlyFinalBestSolutions) {
+                jobs.add(solverManager.solve(
+                        id,
+                        problemId -> PlannerTestUtils.generateTestdataSolution(solutionName, 2),
+                        consumedBestSolutions::add, null));
+            } else {
+                jobs.add(solverManager.solveAndListen(
+                        id,
+                        problemId -> PlannerTestUtils.generateTestdataSolution(solutionName, 2),
+                        consumedBestSolutions::add, null));
+            }
+            consumedSolutions.put(id, consumedBestSolutions);
+        }
+        assertInitializedJobs(jobs);
+        assertConsumedSolutions(problemCount, onlyFinalBestSolutions, consumedSolutions);
+    }
+
+    private void assertConsumedSolutions(int problemCount, boolean onlyFinalBestSolutions, Map<Integer, List<TestdataSolution>> consumedSolutions) {
+        if (onlyFinalBestSolutions) {
+            assertConsumedFinalBestSolutions(problemCount, consumedSolutions);
+        } else {
+            assertConsumedBestSolutions(problemCount, consumedSolutions);
+        }
+    }
+
+    private void assertConsumedFinalBestSolutions(int problemCount, Map<Integer, List<TestdataSolution>> consumedSolutions) {
+        for (int i = 0; i < problemCount; i++) {
+            assertThat(consumedSolutions.get(i)).hasSize(1);
+            TestdataSolution solution = consumedSolutions.get(i).get(0);
+            assertConsumedFinalBestSolution(solution);
+        }
+    }
+
+    private void assertConsumedFinalBestSolution(TestdataSolution solution) {
+        TestdataEntity entity = solution.getEntityList().get(0);
+        assertThat(entity.getCode()).isEqualTo("e1");
+        assertThat(entity.getValue().getCode()).isEqualTo("v1");
+        entity = solution.getEntityList().get(1);
+        assertThat(entity.getCode()).isEqualTo("e2");
+        assertThat(entity.getValue().getCode()).isEqualTo("v2");
+    }
+
+    private void assertConsumedBestSolutions(int problemCount, Map<Integer, List<TestdataSolution>> consumedSolutions) {
+        for (int i = 0; i < problemCount; i++) {
+            assertThat(consumedSolutions.get(i)).hasSize(2);
+            assertConsumedFirstBestSolution(consumedSolutions.get(i).get(0));
+            assertConsumedFinalBestSolution(consumedSolutions.get(i).get(1));
+        }
+    }
+
+    private void assertConsumedFirstBestSolution(TestdataSolution solution) {
+        TestdataEntity entity = solution.getEntityList().get(0);
+        assertThat(entity.getCode()).isEqualTo("e1");
+        assertThat(entity.getValue().getCode()).isEqualTo("v1");
+        entity = solution.getEntityList().get(1);
+        assertThat(entity.getCode()).isEqualTo("e2");
+        assertThat(entity.getValue()).isNull();
+    }
+
+    @Test(timeout = 100)
```

Comment:
I wouldn't take decreasing timeouts to extreme. 100 ms could be too little. Units to tens of seconds is the most usual choice.